### PR TITLE
[FD-1504] Add standard terminology to a Terminology Collection

### PR DESCRIPTION
**Pull request by brendagutman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -554,6 +554,33 @@ This is the expected result after removing preferences with the URL code specify
 }
 ```
 
+### https://[APPURL]/api/Terminology/[id]/standard
```
Comment (brendagutman) [inline README.md:557]: Not sure what we want the endpoint to be.

Also,
I use `standard_terminology` to define the references throughout this PR.
Let me know if we wanted different language surrounding this entirely. Should `standard_terminology` be something else (`prefered_terminology`) ?

File: src/locutus/model/terminology.py [hunk 2]
```diff
@@ -572,6 +572,76 @@ def remove_pref(self, code=None):
...
+    def get_preferred_terminology(self):
+        """
+        Retrieves all references from the 'preferred_terminology' sub-collection
+
+        Returns:
+            dict: An array of `Terminology` references under the "references" key
+        
+        Example:
+        {
+            "reference": [
```
Comment (torstees) [inline src/locutus/model/terminology.py:584]: Let's change this to 
[
  { 
     "reference": "Terminology/tm--asdfasdf"
  },
  { 
    "reference": "Terminology/tm-fdsa-fdsa"
  }
]

If you want to return an object, then you could name the array you return. Also, those reference strings will not start with a slash. It's just {ResourceType}/{id} -- blatant stealing from FHIR.

File: src/locutus/api/terminology.py [hunk 3]
```diff
@@ -214,4 +214,29 @@ def delete(self, id, code=None):
...
+        body = request.get_json()
+        t = Term.get(id)
+        if "preferred_terminology" not in body:
+            return {"message": "preferred_terminology is required"}, 400
+
+        preferred_terminology = body["preferred_terminology"]
+
+        t.add_preferred_terminology(preferred_terminology=preferred_terminology)
+        response = {
+            "terminology": {"Reference being mapped": f"Terminology/{t.id}"},
```
Comment (torstees) [inline src/locutus/api/terminology.py:238]: I'm inclined to think this should be:
response = {
  "id": t.id,
  "preferred_terminology": preferred_terminology,
}
If you want to return the reference details for the terminology you are changing, you should make it a formal reference (so "Reference being mapped" would be "reference"). But, I think id is fine.

File: src/locutus/model/terminology.py [hunk 4]
```diff
@@ -572,6 +572,76 @@ def remove_pref(self, code=None):
...
+            doc_snapshot = doc_ref.get()
+            if doc_snapshot.exists:
+                data = doc_snapshot.to_dict()
+                references = data.get("references", {})
+            else:
+                references = {}
+
+            # Create a new reference(unique key required - python)
+            key = f"reference_{len(references) + 1}"
+            new_pref = {key: f"/Terminology/{preferred_terminology}"}
```
Comment (torstees) [inline src/locutus/model/terminology.py:633]: If these are actual reference strings, that "/" at the beginning doesn't need to be there.

File: README.md [hunk 5]
```diff
@@ -554,6 +554,33 @@ This is the expected result after removing preferences with the URL code specify
...
+
+#### GET
+
+Return the reference to the preferred-terminology related to the `Terminology` (specified by id).<br>
+Returns the preferred-terminology at the Terminology("self") level. <br> Expected return example below.
+
+```json
+{
+    "references": {
+        "reference_1": "/Terminology/tm--example1",
```
Comment (torstees) [inline README.md:567]: Let's break this into two distinct references, one to tm--example1 and the other tm--example2, each as full blown references

"references": [
  { "reference": "Terminology/tm--example1"},
  {"reference": "Terminology/tm--example2"}
]

**Review by torstees — CHANGES_REQUESTED** [review]
This is very close but there are a few suggestions.

**Review by torstees — APPROVED** [review]
looks good to me.

**Pull request merged**